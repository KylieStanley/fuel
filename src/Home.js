import React from 'react'
import Search from './Search'
import Menu	 from './Menu'
import CardContainer from './containers/CardContainer'

export const Home = (props) => {

	return (
		<div className="main-container">
		<Menu />
			<h1 className="header">fuel<span>{props.itemType}</span></h1>
			<CardContainer itemType={props.itemType}/>
		</div>
	)
}

export default Home