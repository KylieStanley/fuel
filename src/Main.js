import React from 'react'

const Main = () => {
	return (
		<div className="splash">
			<div className="ovrlay"></div>
			<div className="splash-green"></div>
				<div className="main-text">
					<h1 className="fuel">fuel</h1>
					<h3>Recipes and Nutrition to support your training</h3>
				</div>
		</div>	
	)
}

export default Main