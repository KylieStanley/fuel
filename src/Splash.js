import React from 'react'
import './styles/main.scss';
import { Link } from 'react-router-dom'


const Splash = () => {
	return (
		<div className="splash-text">
			<h1>fuel</h1>
			<h2>Recipes and Nutrition to support your training</h2>
			<Link to='/main/'>
				<button className="splash-btn">Find Your Fuel <i className="fas fa-caret-right"></i></button>
			</Link>
		</div>
	)	
}

export default Splash