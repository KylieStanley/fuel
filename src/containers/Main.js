import React, { Component } from 'react'
import Form from './Form'
import CardContainer from '../containers/CardContainer'
import { fetchRecipes } from '../thunks/fetchRecipes'
import { connect } from 'react-redux'
import { apiId, apiKey } from '../apiKey'
import Menu from '../components/Menu'
import PropTypes from 'prop-types'


export class Main extends Component {

	componentDidMount() {
		if (this.props.recipes.length === 0) {
			let url = `https://api.edamam.com/search?q=ma&app_id=${apiId}&app_key=${apiKey}&from=0&to=30`
			this.props.fetchRecipes(url)	
		}
	}

	render() {
		return (
			<div className="main-container">
			<div className="header-menu">
				<div className="main-text">
					<h1 className="fuel">fuel</h1>
					<h3>Recipes and Nutrition to support your training</h3>
				</div>
				<Menu />
				</div>
				<Form />
				<CardContainer />
			</div>
		)
	}
}

export const mapStateToProps = (state) => ({
	recipes: state.recipes
})

export const mapDispatchToProps = (dispatch) => ({
  fetchRecipes: (url) => dispatch(fetchRecipes(url))
})

Main.propTypes = {
	recipes: PropTypes.array.isRequired,
	fetchRecipes: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Main)
