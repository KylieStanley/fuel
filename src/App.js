import React, { Component } from 'react';
import './styles/main.scss';
import { Route } from 'react-router-dom'
import Splash from './Splash'
import Home from './Home'
import Main from './containers/Main'


class App extends Component {

  render() {
    return (
    <div className="splash">
      <div className="overlay"></div>
      <div className="splash-green"></div>
        <Route exact path='/' component={ Splash } />
        <Route exact path='/main' component={ Main } />
        <Route exact path='/home' component={ Home } />
      </div>
    );
  }
}

export default App;
