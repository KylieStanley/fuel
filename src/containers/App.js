import React, { Component } from 'react';
import '../styles/main.scss';
import { Route, withRouter } from 'react-router-dom'
import Splash from '../components/Splash'
import List from '../components/List'
import { CardModal } from '../components/CardModal'
import Main from './Main'
import { ModalContainer, ModalRoute } from 'react-router-modal';
import { connect } from 'react-redux'
import 'react-router-modal/css/react-router-modal.css'


export const App = (props) => {
  return (
    <div className="splash">
      <div className="overlay" />
      <div className="splash-green" />
      <Route exact path='/' component={ Splash } />
      <Route path='/main' component={ Main } />
      <Route path="/favorites" render={() => <List itemType="favorites" />} />
      <Route path="/shopping" render={() => <List itemType="shopping" />} />
      <ModalRoute component={ CardModal } path='*/modal' parentPath={`${props.history.goBack}`} props={{ recipe: props.selectedCard }} />
      <ModalContainer />
    </div>
  );
}

export const mapStateToProps = (state) => ({
  selectedCard: state.selectedCard
})

export default withRouter(connect(mapStateToProps)(App));

