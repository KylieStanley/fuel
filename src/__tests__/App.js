import React from 'react'
import { Home } from '../Home'
import  Main  from '../containers/Main'
import App from '../App'
import Splash from '../Splash'
import { shallow, mount } from 'enzyme'
import { Route, component, render } from 'react-router-dom'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'


jest.mock('../containers/CardContainer')
jest.mock('../containers/Main',() => () => <div />)


describe('App', () => {
	let wrapper

	beforeEach(() => {
		wrapper = shallow(<App />)
	})

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should render four possible routes', () => {
    expect(wrapper.find(Route).length).toEqual(4)
  })

  describe('routes', () => {
    let mockStore
    let mockRecipes
    let mockFavorites

    beforeEach(() => {
      mockRecipes = [{ name: 'chicken' }, { name: 'salmon' }]
      mockFavorites = [{ name: 'chicken' }]
      mockStore = {getState: jest.fn(), subscribe: jest.fn(), recipes: mockRecipes, favorites: mockRecipes, dispatch: jest.fn()} 
    })

    it('routes /favorites to the Home component', () => {
      const wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/favorites']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Home).length).toEqual(1)
    })

    it('routes /shopping to the Home component', () => {
      const wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/shopping']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Home).length).toEqual(1)
    })

    it('routes /main to the Main component', () => {
      const wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/main']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Main).length).toEqual(1)
    })

    it('routes / to the Splash component', () => {
      const wrapper = mount(
        <Provider store={mockStore}>
          <MemoryRouter initialEntries={['/']}>
            <App />
          </MemoryRouter>
        </Provider>
      )
      expect(wrapper.find(Splash).length).toEqual(1)
    })
  })
})
