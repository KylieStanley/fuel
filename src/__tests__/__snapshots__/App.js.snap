// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="overlay"
        />,
        <div
          className="splash-green"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/main"
        />,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/shopping"
          render={[Function]}
        />,
      ],
      "className": "splash",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "overlay",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "splash-green",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/main",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/shopping",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="overlay"
          />,
          <div
            className="splash-green"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/main"
          />,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/shopping"
            render={[Function]}
          />,
        ],
        "className": "splash",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "overlay",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "splash-green",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/main",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/shopping",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
